spacy/cli/debug_diff.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional
+
+import typer
+from wasabi import Printer, diff_strings, MarkdownRenderer
+from pathlib import Path
+from thinc.api import Config
+
+from ._util import debug_cli, Arg, Opt, show_validation_error, parse_config_overrides
+from ..util import load_config
+from .init_config import init_config, Optimizations
+
+
+@debug_cli.command(
+    "diff",
+    context_settings={"allow_extra_args": True, "ignore_unknown_options": True},
+)
+def debug_diff_cli(
+    # fmt: off
+    ctx: typer.Context,
+    config_path: Path = Arg(..., help="Path to config file", exists=True, allow_dash=True),
+    compare_to: Optional[Path] = Opt(None, help="Path to another config file to diff against", exists=True, allow_dash=True),
+    optimize: Optimizations = Opt(Optimizations.efficiency.value, "--optimize", "-o", help="Whether the user config was optimized for efficiency or accuracy."),
+    gpu: bool = Opt(False, "--gpu", "-G", help="Whether the original config can run on a GPU"),
+    pretraining: bool = Opt(False, "--pretraining", "--pt", help="Whether to compare on a config with pretraining involved"),
```

Comment:
The `optimize`, `gpu` and `pretraining` options are only relevant when `compare_to` is `None`, right? So maybe we should document more clearly that these only affect the "default" config that would be compared to (both here as well as in the docs).

Reply:
UPdated here: https://github.com/explosion/spaCy/pull/10502/commits/ffdb3a017d99bc2e219a088186eea8d0f3c5265f